fix unit test failures on Windows

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->


**What happened**:

following unit test could not pass on Windows, we should either fix those failures on Windows or disable those tests running on Windows
```
C:\Users\xiazhang\go\src\github.com\kubernetes-csi\csi-driver-smb\pkg\smb>go test
E0812 10:32:35.859486 1580532 nodeserver.go:313] MakeDir failed on target: ./smb.go (mkdir ./smb.go: The system cannot find the path specified.)
--- FAIL: TestNodeStageVolume (0.04s)
    nodeserver_test.go:122: test case: [Error] Not a Directory, Unexpected error: rpc error: code = Internal desc = Could not mount target "./smb.go": mkdir ./smb.go: The system cannot find the path specified.
    nodeserver_test.go:122: test case: [Error] Failed SMB mount mocked by MountSensitive, Unexpected error: prepare stage path failed for ./error_mount_sens_source with error: could not cast to csi proxy class
    nodeserver_test.go:122: test case: [Success] Valid request, Unexpected error: prepare stage path failed for ./source_test with error: could not cast to csi proxy class
E0812 10:32:35.915490 1580532 nodeserver.go:313] MakeDir failed on target: ./smb.go (mkdir ./smb.go: The system cannot find the path specified.)
--- FAIL: TestNodePublishVolume (0.02s)
    nodeserver_test.go:274: test case: [Error] Not a directory, Unexpected error: rpc error: code = Internal desc = Could not mount target "./smb.go": mkdir ./smb.go: The system cannot find the path specified.
    nodeserver_test.go:274: test case: [Error] Mount error mocked by Mount, Unexpected error: prepare publish failed for ./target_test with error: could not cast to csi proxy class
    nodeserver_test.go:274: test case: [Success] Valid request read only, Unexpected error: prepare publish failed for ./target_test with error: could not cast to csi proxy class
    nodeserver_test.go:274: test case: [Success] Valid request, Unexpected error: prepare publish failed for ./target_test with error: could not cast to csi proxy class
--- FAIL: TestNodeUnpublishVolume (0.00s)
    nodeserver_test.go:327: test case: [Error] Unmount error mocked by IsLikelyNotMountPoint, Unexpected error: rpc error: code = Internal desc = failed to unmount target "./error_is_likely_target": could not cast to csi proxy class
    nodeserver_test.go:327: test case: [Success] Valid request, Unexpected error: rpc error: code = Internal desc = failed to unmount target "./abc.go": could not cast to csi proxy class
--- FAIL: TestNodeUnstageVolume (0.00s)
    nodeserver_test.go:378: test case: [Error] CleanupMountPoint error mocked by IsLikelyNotMountPoint, Unexpected error: rpc error: code = Internal desc = failed to unmount staging target "./error_is_likely_target": could not cast to csi proxy class
    nodeserver_test.go:378: test case: [Success] Valid request, Unexpected error: rpc error: code = Internal desc = failed to unmount staging target "./abc.go": could not cast to csi proxy class
IsCorruptedDir(./error_is_likely_target) returned with error: <nil>W0812 10:32:35.983487 1580532 nodeserver.go:303] ReadDir ./false_is_likely_target failed with open ./false_is_likely_target: The system cannot find the file specified., unmount this directory
E0812 10:32:35.984504 1580532 nodeserver.go:313] MakeDir failed on target: ./smb.go (mkdir ./smb.go: The system cannot find the path specified.)
--- FAIL: TestIsCorruptedDir (0.00s)
    smb_test.go:54: failed to create curruptedPath: symlink C:\Users\xiazhang\AppData\Local\Temp\csi-mount-test216866723 C:\Users\xiazhang\AppData\Local\Temp\csi-mount-test216866723\curruptedPath: A required privilege is not held by the client.
I0812 10:32:36.012492 1580532 smb.go:58]
DRIVER INFORMATION:
-------------------
Build Date: N/A
Compiler: gc
Driver Name: smb.csi.k8s.io
Driver Version: N/A
Git Commit: N/A
Go Version: go1.13.14
```
**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/pkg/smb/nodeserver_test.go
+++ b/pkg/smb/nodeserver_test.go
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"os"
 	"reflect"
+	"runtime"
 	"syscall"
 	"testing"
 
@@ -37,7 +38,16 @@ const (
 	targetTest = "./target_test"
 )
 
+// checks if the underlying OS is Windows
+func isWindows() bool {
+	return runtime.GOOS == "windows"
+}
+
 func TestNodeStageVolume(t *testing.T) {
+	if isWindows() {
+		t.Skip("Skipping tests on Windows")
+	}
+
 	stdVolCap := csi.VolumeCapability{
 		AccessType: &csi.VolumeCapability_Mount{
 			Mount: &csi.VolumeCapability_MountVolume{},
@@ -180,6 +190,10 @@ func TestNodeExpandVolume(t *testing.T) {
 }
 
 func TestNodePublishVolume(t *testing.T) {
+	if isWindows() {
+		t.Skip("Skipping tests on Windows")
+	}
+
 	volumeCap := csi.VolumeCapability_AccessMode{Mode: csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER}
 	errorMountSource := "./error_mount_source"
 	alreadyMountedTarget := "./false_is_likely_exist_target"
@@ -283,6 +297,10 @@ func TestNodePublishVolume(t *testing.T) {
 }
 
 func TestNodeUnpublishVolume(t *testing.T) {
+	if isWindows() {
+		t.Skip("Skipping tests on Windows")
+	}
+
 	errorTarget := "./error_is_likely_target"
 	targetFile := "./abc.go"
 
@@ -334,6 +352,10 @@ func TestNodeUnpublishVolume(t *testing.T) {
 }
 
 func TestNodeUnstageVolume(t *testing.T) {
+	if isWindows() {
+		t.Skip("Skipping tests on Windows")
+	}
+
 	errorTarget := "./error_is_likely_target"
 	targetFile := "./abc.go"
 

--- a/pkg/smb/nodeserver_test.go
+++ b/pkg/smb/nodeserver_test.go
@@ -38,15 +38,15 @@ const (
 	targetTest = "./target_test"
 )
 
-// checks if the underlying OS is Windows
-func isWindows() bool {
-	return runtime.GOOS == "windows"
+// skip test if running on windows
+func skipTestIfWindows(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test on Windows")
+	}
 }
 
 func TestNodeStageVolume(t *testing.T) {
-	if isWindows() {
-		t.Skip("Skipping tests on Windows")
-	}
+	skipTestIfWindows(t)
 
 	stdVolCap := csi.VolumeCapability{
 		AccessType: &csi.VolumeCapability_Mount{
@@ -190,10 +190,7 @@ func TestNodeExpandVolume(t *testing.T) {
 }
 
 func TestNodePublishVolume(t *testing.T) {
-	if isWindows() {
-		t.Skip("Skipping tests on Windows")
-	}
-
+	skipTestIfWindows(t)
 	volumeCap := csi.VolumeCapability_AccessMode{Mode: csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER}
 	errorMountSource := "./error_mount_source"
 	alreadyMountedTarget := "./false_is_likely_exist_target"
@@ -297,10 +294,7 @@ func TestNodePublishVolume(t *testing.T) {
 }
 
 func TestNodeUnpublishVolume(t *testing.T) {
-	if isWindows() {
-		t.Skip("Skipping tests on Windows")
-	}
-
+	skipTestIfWindows(t)
 	errorTarget := "./error_is_likely_target"
 	targetFile := "./abc.go"
 
@@ -352,10 +346,7 @@ func TestNodeUnpublishVolume(t *testing.T) {
 }
 
 func TestNodeUnstageVolume(t *testing.T) {
-	if isWindows() {
-		t.Skip("Skipping tests on Windows")
-	}
-
+	skipTestIfWindows(t)
 	errorTarget := "./error_is_likely_target"
 	targetFile := "./abc.go"
 

--- a/pkg/smb/smb_test.go
+++ b/pkg/smb/smb_test.go
@@ -43,6 +43,10 @@ func TestNewFakeDriver(t *testing.T) {
 }
 
 func TestIsCorruptedDir(t *testing.T) {
+	if isWindows() {
+		t.Skip("Skipping tests on windows")
+	}
+
 	existingMountPath, err := ioutil.TempDir(os.TempDir(), "csi-mount-test")
 	if err != nil {
 		t.Fatalf("failed to create tmp dir: %v", err)

--- a/pkg/smb/smb_test.go
+++ b/pkg/smb/smb_test.go
@@ -43,10 +43,7 @@ func TestNewFakeDriver(t *testing.T) {
 }
 
 func TestIsCorruptedDir(t *testing.T) {
-	if isWindows() {
-		t.Skip("Skipping tests on windows")
-	}
-
+	skipTestIfWindows(t)
 	existingMountPath, err := ioutil.TempDir(os.TempDir(), "csi-mount-test")
 	if err != nil {
 		t.Fatalf("failed to create tmp dir: %v", err)

